app/ante/eip712_test.go

```diff
@@ -0,0 +1,611 @@
+			errMsg:         "transfer amount exceeds balance",
+		},
+		{
+			name:           "fails when minting more usdx than allowed",
+			usdcDepositAmt: 100,
+			usdxToMintAmt:  100,
+			errMsg:         "proposed collateral ratio is below liquidation ratio",
+		},
+		{
+			name:           "fails when trying to convert usdc for another address",
+			usdcDepositAmt: 100,
+			usdxToMintAmt:  90,
+			errMsg:         "unauthorized",
+			failCheckTx:    true,
+			updateMsgs: func(msgs []sdk.Msg) []sdk.Msg {
+				bridgeMsg := bridgetypes.NewMsgConvertERC20ToCoin(
+					suite.testEVMAddr2,
+					suite.testAddr,
+					suite.usdcEVMAddr,
+					suite.getEVMAmount(100),
+				)
+				msgs[0] = &bridgeMsg
+				return msgs
+			},
+		},
+		{
+			name:           "fails when trying to convert erc20 for non-whitelisted contract",
+			usdcDepositAmt: 100,
+			usdxToMintAmt:  90,
+			errMsg:         "ERC20 token not enabled to convert to sdk.Coin",
+			updateMsgs: func(msgs []sdk.Msg) []sdk.Msg {
+				bridgeMsg := bridgetypes.NewMsgConvertERC20ToCoin(
+					suite.testEVMAddr,
+					suite.testAddr,
+					suite.testEVMAddr2,
+					suite.getEVMAmount(100),
+				)
+				msgs[0] = &bridgeMsg
+				return msgs
+			},
+		},
+		{
+			name:           "fails when signer tries to send messages with invalid signature",
+			usdcDepositAmt: 100,
+			usdxToMintAmt:  90,
+			failCheckTx:    true,
+			errMsg:         "tx intended signer does not match the given signer",
+			updateTx: func(txBuilder client.TxBuilder, msgs []sdk.Msg) client.TxBuilder {
+				var option *codectypes.Any
+				option, _ = codectypes.NewAnyWithValue(&etherminttypes.ExtensionOptionsWeb3Tx{
+					FeePayer:         suite.testAddr.String(),
+					TypedDataChainID: 1,
+					FeePayerSig:      []byte("sig"),
+				})
+				builder, _ := txBuilder.(authtx.ExtensionOptionsTxBuilder)
+				builder.SetExtensionOptions(option)
+				return txBuilder
+			},
+		},
+		{
+			name:           "fails when insufficient gas fees are provided",
+			usdcDepositAmt: 100,
+			usdxToMintAmt:  90,
+			errMsg:         "insufficient funds",
+			updateTx: func(txBuilder client.TxBuilder, msgs []sdk.Msg) client.TxBuilder {
+				bk := suite.tApp.GetBankKeeper()
+				gasCoins := bk.GetBalance(suite.ctx, suite.testAddr, "ukava")
+				suite.tApp.GetBankKeeper().SendCoins(suite.ctx, suite.testAddr, suite.testAddr2, sdk.NewCoins(gasCoins))
+				return txBuilder
+			},
+		},
+		{
+			name:           "fails when invalid chain id is provided",
+			usdcDepositAmt: 100,
+			usdxToMintAmt:  90,
+			failCheckTx:    true,
+			errMsg:         "invalid chain-id",
+			updateTx: func(txBuilder client.TxBuilder, msgs []sdk.Msg) client.TxBuilder {
+				gasAmt := sdk.NewCoins(sdk.NewCoin("ukava", sdk.NewInt(20)))
+				return suite.createTestEIP712CosmosTxBuilder(
+					suite.testAddr, suite.testPrivKey, "kavatest_12-1", uint64(helpers.DefaultGenTxGas*2), gasAmt, msgs,
+				)
+			},
+		},
+		{
+			name:           "fails when invalid fee payer is provided",
```

Comment:
here are the fail cases

Reply:
it's pretty important to make sure this covers all the critical error cases. Thoughts on any additional error cases that should be added here? @rhuairahrighairidh @karzak @nddeluca @drklee3